dns/pfSense-pkg-bind9/files/usr/local/pkg/bind.inc

```diff
@@ -739,113 +739,128 @@ EOD;
+					$("input#retry").attr("disabled", true);
+					$("input#expire").attr("disabled", true);
+					$("input#minimum").attr("disabled", true);
+					$("select#allowquery\\[\\]").attr("disabled", true);
+					$("select#allowupdate\\[\\]").attr("disabled", true);
+					$("select#allowtransfer\\[\\]").attr("disabled", true);
+					$("input#enable_updatepolicy").attr("disabled", true);
+					$("input#updatepolicy").attr("disabled", true);
+					$("input#rpz").attr("disabled", true);
 					break;
 				case 'redirect':
-					document.iform.slaveip.disabled = 1;
-					document.iform.tll.disabled = 1;
-					document.iform.nameserver.disabled = 0;
-					document.iform.reverso.disabled = 1;
-					document.iform.reversv6o.disabled = 1;
-					document.iform.forwarders.disabled = 1;
-					document.iform.dnssec.disabled = 1;
-					document.iform.backupkeys.disabled = 1;
-					document.iform.regdhcpstatic.disabled = 1;
-					document.iform.ipns.disabled = 1;
-					document.iform.mail.disabled = 0;
-					document.iform.serial.disabled = 0;
-					document.iform.refresh.disabled = 0;
-					document.iform.retry.disabled = 0;
-					document.iform.expire.disabled = 0;
-					document.iform.minimum.disabled = 0;
-					document.iform.allowquery.disabled = 0;
-					document.iform.allowupdate.disabled = 1;
-					document.iform.enable_updatepolicy.disabled = 1;
-					document.iform.updatepolicy.disabled = 1;
+					$("input#slaveip").attr("disabled", true);
+					$("input#ttl").attr("disabled", true);
+					$("input#nameserver").attr("disabled", false);
+					$("input#reverso").attr("disabled", true);
+					$("input#reversv6o").attr("disabled", true);
+					$("input#forwarders").attr("disabled", false);
+					$("input#dnssec").attr("disabled", true);
+					$("input#backupkeys").attr("disabled", true);
+					$("input#regdhcpstatic").attr("disabled", true);
+					$("input#ipns").attr("disabled", true);
+					$("input#mail").attr("disabled", false);
+					$("input#serial").attr("disabled", false);
+					$("input#refresh").attr("disabled", false);
+					$("input#retry").attr("disabled", false);
+					$("input#expire").attr("disabled", false);
+					$("input#minimum").attr("disabled", false);
+					$("select#allowquery\\[\\]").attr("disabled", false);
+					$("select#allowupdate\\[\\]").attr("disabled", true);
+					$("select#allowtransfer\\[\\]").attr("disabled", true);
+					$("input#enable_updatepolicy").attr("disabled", true);
+					$("input#updatepolicy").attr("disabled", true);
+					$("input#rpz").attr("disabled", true);
 					break;
 				default:
 					break;
 			}
 		}
-		-->
 	</script>
-	<?php
+JS;
+	echo $js;
 }
 
 function bind_print_javascript_type_zone2() {
-	print("<script language=\"text/javascript\">on_type_zone_changed(); document.iform.resultconfig.disabled = 1; document.iform.dsset.disabled = 1;</script>\n");
+	$js = <<<'JS'
+	<script>
+	// we can't put this at the end of the body, so wait until jQuery is loaded
```

Comment:
Elsewhere throughout pfSense we use events.push() to ensure JS is executed after the include files (jQuery etc) are loaded. Would that work here? I would be better to use a single approach.

Here is a small example:

`<script type="text/javascript">
//<![CDATA[
events.push(function() {
	// Has the Use ramdisk checkbox changed state?
	if (<?=(int)$use_mfs_tmpvar_changed?> && confirm("<?=$ramdisk_msg?>")) {
		postSubmit({override : 'yes'}, 'diag_reboot.php')
	}

	// source track timeout field is disabled if sticky connections not enabled
	$('#lb_use_sticky').click(function () {
		disableInput('srctrack', !$(this).prop("checked"));
	});

	disableInput('srctrack', !$('#lb_use_sticky').prop("checked"));

});
//]]>
</script>`

Reply:
Updated to use `events.push()`; also consolidated the script into one block and got rid of ugly 1990's `onchange` attributes!